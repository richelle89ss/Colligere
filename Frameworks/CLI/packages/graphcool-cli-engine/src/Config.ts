import { RunOptions } from './types/common'
import * as path from 'path'
import * as os from 'os'
import * as fs from 'fs-extra'
import * as cuid from 'scuid'
import * as findUp from 'find-up'
import { Output } from './Output/index'
import * as yaml from 'js-yaml'
const debug = require('debug')('config')


export class Config {
  /**
   * Local settings
   */
  out: Output
  debug: boolean = Boolean(
    process.env.DEBUG && process.env.DEBUG!.includes('*'),
  )
  windows: boolean = false
  bin: string = 'graphcool'
  mock: boolean = true
  argv: string[] = process.argv.slice(1)
  commandsDir: string = path.join(__dirname, '../dist/commands')
  defaultCommand: string = 'help'
  userPlugins: boolean = false
  version: string = '1.3.11'
  name: string = 'graphcool'
  pjson: any = {
    name: 'cli-engine',
    version: '0.0.0',
    dependencies: {},
    'cli-engine': {
      defaultCommand: 'help',
    },
  }
  sharedClusters: string[] = ['shared-eu-west-1', 'shared-ap-northeast-1', 'shared-us-west-2']

  /**
   * Paths
   */
  cwd: string
  home: string
  root = path.join(__dirname, '..')

  definitionDir: string
  definitionPath: string | null
  localRCPath: string
  globalRCPath: string
  warnings: string[] = []

  /**
   * Urls
   */
  authUIEndpoint = process.env.ENV === 'DEV'
    ? 'https://dev.console.graph.cool/cli/auth'
    : 'https://console.graph.cool/cli/auth'
  backendAddr = process.env.ENV === 'DEV'
    ? 'https://dev.api.graph.cool'
    : 'https://api.graph.cool'
  systemAPIEndpoint = process.env.ENV === 'DEV'
    ? 'https://dev.api.graph.cool/system'
    : 'https://api.graph.cool/system'
  authEndpoint = process.env.ENV === 'DEV'
    ? 'https://cli-auth-api.graph.cool/dev'
    : 'https://cli-auth-api.graph.cool/prod'
  docsEndpoint = process.env.ENV === 'DEV'
    ? 'https://dev.graph.cool/docs'
    : 'https://www.graph.cool/docs'
  statusEndpoint = 'https://crm.graph.cool/prod/status'

  /**
   * consumer endpoints
   */
  simpleAPIEndpoint = process.env.ENV === 'DEV'
    ? 'https://dev.api.graph.cool/simple/v1/'
    : 'https://api.graph.cool/simple/v1/'
  relayAPIEndpoint = process.env.ENV === 'DEV'
    ? 'https://dev.api.graph.cool/relay/v1/'
    : 'https://api.graph.cool/relay/v1/'
  fileAPIEndpoint = process.env.ENV === 'DEV'
    ? 'https://dev.api.graph.cool/file/v1/'
    : 'https://api.graph.cool/file/v1/'
  subscriptionsEndpoint = process.env.ENV === 'DEV'
    ? 'wss://dev.subscriptions.graph.cool'
    : 'wss://subscriptions.graph.cool'


  /* tslint:disable-next-line */
  __cache = {}

  constructor(options?: RunOptions) {
    this.cwd = (options && options.cwd) || this.getCwd()
    this.home = (options && options.home) || this.getHome()
    debug(`CWD`, this.cwd)
    debug(`HOME`, this.home)
    this.setDefinitionPaths()
    this.setRCPaths()
    if (options) {
      this.readPackageJson(options)
    }
  }
  setLocal(host: string = 'http://localhost:60000') {
    this.backendAddr = host
    this.systemAPIEndpoint = host + '/system'
  }
  setOutput(out: Output) {
    this.out = out
    this.warnings.forEach(warning  => out.warn(warning))
    this.warnings = []
  }
  get arch(): string {
    return os.arch() === 'ia32' ? 'x86' : os.arch()
  }
  get platform(): string {
    return os.platform() === 'win32' ? 'windows' : os.platform()
  }
  get userAgent(): string {
    return `${this.name}/${this.version} (${this.platform}-${this
      .arch}) node-${process.version}`
  }
  get dirname() {
    return this.pjson['cli-engine'].dirname || this.bin
  }
  get cacheDir() {
    const x = dir(
      this,
      'cache',
      this.platform === 'darwin'
        ? path.join(this.home, 'Library', 'Caches')
        : null,
    )
    return x
  }
  get requireCachePath() {
    return path.join(this.cacheDir, '/.require-cache.json')
  }
  private readPackageJson(options: RunOptions) {
    this.mock = options.mock
    this.argv = options.argv || this.argv
    if (options.root) {
      this.root = options.root
      const pjsonPath = path.join(options.root, 'package.json')
      const pjson = fs.readJSONSync(pjsonPath)
      if (pjson && pjson['cli-engine']) {
        this.pjson = pjson
        this.version = pjson.version
      }
    }
  }
  private setRCPaths() {
    this.localRCPath = path.join(this.definitionDir, '.graphcoolrc')
    const homePath = path.join(this.home, '.graphcoolrc')
    debug(`homepath`, homePath)
    this.globalRCPath = homePath
    debug(`localRCPath`, this.localRCPath)
    debug(`globalRCPath`, this.globalRCPath)
  }
  private warn(msg: string) {
    this.warnings.push(msg)
  }
  private setDefinitionPaths() {
    const definitionPath = path.join(this.cwd, 'graphcool.yml')
    if (fs.pathExistsSync(definitionPath)) {
      this.definitionDir = this.cwd
      this.definitionPath = definitionPath
    } else {
      const found = findUp.sync('graphcool.yml', {cwd: this.cwd})
      this.definitionDir = found ? path.dirname(found) : this.cwd
      this.definitionPath = found || null
    }
    debug(`definitionDir`, this.definitionDir)
    debug(`definitionPath`, this.definitionPath)
  }
  private getCwd() {
    // get cwd
    let cwd = process.cwd()
    if (process.env.NODE_ENV === 'test') {
      cwd = path.join(os.tmpdir(), `${cuid()}/`)
      fs.mkdirpSync(cwd)
      debug('cwd', cwd)
    }
    return cwd
  }
  private getHome() {
    // get home
    let home = os.homedir() || os.tmpdir()
    if (process.env.NODE_ENV === 'test') {
      home = path.join(os.tmpdir(), `${cuid()}/`)
      fs.mkdirpSync(home)
      debug('home', home)
    }
    return home
  }
}

function dir(config: Config, category: string, d: string | null): string {
  const cacheKey = `dir:${category}`
  const cache = config.__cache[cacheKey]
  if (cache) {
    return cache
  }
  d =
    d ||
    path.join(
      config.home,
      category === 'data' ? '.local/share' : '.' + category,
    )
  if (config.windows) {
    d = process.env.LOCALAPPDATA || d
  }
  d = process.env.XDG_DATA_HOME || d
  d = path.join(d, config.dirname)
  fs.mkdirpSync(d)
  config.__cache[cacheKey] = d
  return d
}
