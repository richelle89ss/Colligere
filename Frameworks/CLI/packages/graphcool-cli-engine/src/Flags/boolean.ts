import { Flag } from './index'

export default function boolean(options: Flag<any> = {}): Flag<boolean> {
  return {
    ...options,
    parse: undefined,
  }
}