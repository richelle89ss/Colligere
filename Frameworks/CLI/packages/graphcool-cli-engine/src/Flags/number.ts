import { Flag } from './index'

export default function number(options: Flag<string> = {}): Flag<number> {
  return {
    ...options,
    parse: (input: string = options.defaultValue) => parseInt(input, 10),
  }
}
