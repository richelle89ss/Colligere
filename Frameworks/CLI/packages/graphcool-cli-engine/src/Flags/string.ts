import { Flag } from './index'

export default function string(options: Flag<string> = {}): Flag<string> {
  return {
    parse: (input: string = options.defaultValue) => input,
    ...options,
  }
}
