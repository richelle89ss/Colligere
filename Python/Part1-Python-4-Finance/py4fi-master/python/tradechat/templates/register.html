{% extends "layout.html" %}
{% block body %}
  <h2>Register</h2>
  {% if error %}<p class=error><strong>Error:</strong> {{ error }}{% endif %}
  <form action="{{ url_for('register') }}" method=post>
    <dl>
      <dd><font size="-1">Username</font>
      <dd><input type=text name=username>
      <dd><font size="-1">Password</font>
      <dd><input type=password name=password>
      <dd><input type=submit value=Register>
    </dl>
  </form>
{% endblock %}