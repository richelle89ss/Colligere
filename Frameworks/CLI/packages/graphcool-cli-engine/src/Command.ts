import 'source-map-support/register'
import { Arg, Flags } from './Flags/index'
import { Output } from './Output'
import { Config } from './Config'
import { ProjectDefinition, RunOptions } from './types/common'
import { OutputArgs, OutputFlags, Parser } from './Parser'
import Help from './Help'
import { Client } from './Client/Client'
import { ProjectDefinitionClass } from './ProjectDefinition/ProjectDefinition'
import { Auth } from './Auth'
import { Environment } from './Environment'
import packagejson = require('../package.json')
import * as mock from './mock'
import * as fs from 'fs-extra'
import * as path from 'path'
import { RC } from './types/rc'
const debug = require('debug')('command')

const pjson = packagejson as any

export class Command {
  static topic: string
  static group: string
  static command?: string
  static description?: string
  static usage?: string
  static flags: Flags
  static args: Arg[] = []
  static aliases: string[] = []
  static hidden: boolean = false
  static mockDefinition: ProjectDefinition
  static mockRC: RC
  static allowAnyFlags: boolean = false

  static get id(): string {
    return this.command ? `${this.topic}:${this.command}` : this.topic
  }

  static async mock(...argv: any[]): Promise<Command> {
    let customArgs: any = null
    if (typeof argv[0] === 'object') {
      customArgs = argv.shift()
    }

    argv.unshift('argv0', 'cmd')

    const mockDefinition =
      customArgs && customArgs.mockDefinition
        ? customArgs.mockDefinition
        : mock.mockDefinition
    const mockRC = customArgs && customArgs.mockRC ? customArgs.mockRC : null
    const mockConfig = customArgs && customArgs.mockConfig ? customArgs.mockConfig : null
    debug(`Using mockDefinition`, mockDefinition)
    debug(`Using mockRC`, mockRC)

    return this.run({ argv, mock: true, mockDefinition, mockRC, mockConfig })
  }

  static async run(config?: RunOptions): Promise<Command> {
    if (process.env.NOCK_WRITE_RESPONSE_CMD === 'true') {
      debug('RECORDING')
      require('nock').recorder.rec({
        dont_print: true,
      })
    }
    const cmd = new this({ config })

    try {
      await cmd.init(config)
      await cmd.run()
      await cmd.out.done()
    } catch (err) {
      cmd.out.error(err)
    }

    if (process.env.NOCK_WRITE_RESPONSE_CMD === 'true') {
      const requests = require('nock').recorder.play()
      const requestsPath = path.join(process.cwd(), 'requests.js')
      debug('WRITING', requestsPath)
      fs.writeFileSync(requestsPath, requests.join('\n'))
    }
    return cmd
  }

  static buildHelp(config: Config): string {
    const help = new Help(config)
    return help.command(this)
  }

  static buildHelpLine(config: Config): string[] {
    const help = new Help(config)
    return help.commandLine(this)
  }

  protected static version = pjson.version

  client: Client
  out: Output
  config: Config
  definition: ProjectDefinitionClass
  auth: Auth
  env: Environment
  flags: OutputFlags
  args?: OutputArgs
  argv: string[]

  constructor(options: { config?: RunOptions } = { config: { mock: true } }) {
    this.config = options.config && options.config.mockConfig || options.config instanceof Config ? (options.config as any) : new Config(options.config)
    this.out = new Output(this.config)
    this.config.setOutput(this.out)
    this.argv = options.config && options.config.argv ? options.config.argv : []
    this.definition = new ProjectDefinitionClass(this.out, this.config)
    this.env = new Environment(this.out, this.config)
    this.client = new Client(this.config, this.env, this.out)
    this.auth = new Auth(this.out, this.config, this.env, this.client)
    this.client.setAuth(this.auth)
  }

  async run(...rest: void[]): Promise<void> {
    // noop
  }

  async init(options?: RunOptions) {
    // parse stuff here
    const mockDefinition = options && options.mockDefinition
    const mockRC = options && options.mockRC
    if (mockDefinition) {
      this.definition.set(mockDefinition)
    }
    if (mockRC) {
      this.env.localRC = mockRC
    }
    const parser = new Parser({
      flags: (this.constructor as any).flags || {},
      args: (this.constructor as any).args || [],
      variableArgs: (this.constructor as any).variableArgs,
      cmd: this,
    })
    if (this.argv && (this.argv.includes('-e') || this.argv.includes('--env'))) {
      this.out.warn('The usage of -e or --env is deprecated. Please use --target instead.')
    }
    const { argv, flags, args } = await parser.parse({
      flags: this.flags,
      argv: this.argv.slice(2),
    })
    this.flags = flags!
    this.argv = argv!
    this.args = args
    await this.env.load(flags!)
  }

  get stdout(): string {
    return this.out.stdout.output
  }

  get stderr(): string {
    return this.out.stderr.output
  }
}
