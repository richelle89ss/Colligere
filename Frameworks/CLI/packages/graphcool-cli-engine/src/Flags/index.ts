import { Command } from '../Command'
import boolean from './boolean'
import string from './string'
import number from './number'

export interface Flags {[name: string]: Flag<any>}

export type AlphabetUppercase = | 'A' | 'B' | 'C' | 'D' | 'E' | 'F' | 'G' | 'H' | 'I' | 'J' | 'K' | 'L' | 'M' | 'N' | 'O' | 'P' | 'Q' | 'R' | 'S' | 'T' | 'U' | 'V' | 'W' | 'X' | 'Y' | 'Z'
export type AlphabetLowercase = | 'a' | 'b' | 'c' | 'd' | 'e' | 'f' | 'g' | 'h' | 'i' | 'j' | 'k' | 'l' | 'm' | 'n' | 'o' | 'p' | 'q' | 'r' | 's' | 't' | 'u' | 'v' | 'w' | 'x' | 'y' | 'z'

export interface Flag<T> {
  parse?: (flag?: string, cmd?: Command, name?: string) => T
  char?: AlphabetLowercase | AlphabetUppercase
  description?: string
  hidden?: boolean
  required?: boolean
  defaultValue?: any
}

export interface Arg {
  name: string
  description?: string
  required?: boolean
  hidden?: boolean
}

export const flags = {boolean, string, number}
