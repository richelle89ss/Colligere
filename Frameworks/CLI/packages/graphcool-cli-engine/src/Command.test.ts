/* tslint:disable */
import 'source-map-support/register'

import {Command as Base} from './Command'
import {flags as Flags} from './Flags'
import nock from 'nock'
import { Config } from './Config'

class Command extends Base {
  static topic = 'foo'
  static command = 'bar'
  static flags = {myflag: Flags.boolean()}
  static args = [{name: 'myarg', required: false}]
}

test('shows the ID', () => {
  expect(Command.id).toEqual('foo:bar')
})

test('runs the command', async () => {
  const cmd = await Command.mock()
  expect(cmd.flags).toEqual({})
  expect(cmd.argv).toEqual([])
})

test('has stdout', async () => {
  class Command extends Base {
    static flags = {
      print: Flags.string(),
      bool: Flags.boolean()
    }
    async run () {
      this.out.stdout.log(this.flags.print)
    }
  }

  const {stdout} = await Command.mock('--print=foo')
  expect(stdout).toEqual('foo\n')
})

test('has stderr', async () => {
  class Command extends Base {
    static flags = {print: Flags.string()}
    async run () {
      this.out.stderr.log(this.flags.print)
    }
  }

  const {stderr} = await Command.mock('--print=foo')
  expect(stderr).toEqual('foo\n')
})

test('parses args', async () => {
  const cmd = await Command.mock('one')
  expect(cmd.flags).toEqual({})
  expect(cmd.argv).toEqual(['one'])
  expect(cmd.args).toEqual({myarg: 'one'})
})

test('has help', async () => {
  class Command extends Base {
    static topic = 'config'
    static command = 'get'
    static help = `this is

some multiline help
`
  }
  let config = new Config({mock: true})
  expect(Command.buildHelp(config)).toMatchSnapshot('buildHelp')
  expect(Command.buildHelpLine(config)).toMatchSnapshot('buildHelpLine')
})
