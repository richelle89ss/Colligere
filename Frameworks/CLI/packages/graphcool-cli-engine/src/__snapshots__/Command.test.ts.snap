// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildHelp 1`] = `
"Usage: graphcool config get

this is

some multiline help
"
`;

exports[`buildHelpLine 1`] = `
Array [
  "config get",
  null,
]
`;
